src/Conversion/ONNXToStablehlo/Tensor/Gather.cpp

```diff
@@ -56,19 +57,16 @@ struct ONNXGatherOpLoweringToStablehlo : public ConversionPattern {
     // start indices
     Value zero = getShapedZero(loc, rewriter, indices);
     Value axisDimSize;
-    if (inputType.hasStaticShape()) {
+    if (!inputType.isDynamicDim(axisLit)) {
       int64_t axisDimSizeLit = inputType.getShape()[axisLit];
       axisDimSize = getShapedInt(loc, rewriter, axisDimSizeLit, indices);
     } else {
-      Value inputShape = rewriter.create<shape::ShapeOfOp>(loc, data);
-      Value indicesShape = rewriter.create<shape::ShapeOfOp>(loc, indices);
-      Value axisDimSizeIndexValue =
-          rewriter.create<shape::GetExtentOp>(loc, inputShape, axisLit);
-      Value axisDimSizeValue = rewriter.create<arith::IndexCastOp>(
-          loc, indicesType.getElementType(), axisDimSizeIndexValue);
```

Comment:
the issue was that `axisDimSizeValue` is an `i64` not a `tensor<i64>` i could just use the old method but add a `tensor.from_elements` to get the right type. but i thought `tensor.dim` + `tensor.splat` was more straight forward. however if this is against design principles I can do the former.